### PR TITLE
For #4991, #4993:Ensured deleteSelectedBookmarks deletes all selected bookmarks 

### PR DESCRIPTION
**Pull request by mcarare**
Moved the deletion to a CoroutineScope(IO) to decouple from fragment lifecycle
Used local val tempStorage to keep reference to bookmarkStorage of outside class
Ran deletions in a runBlocking to avoid concurrency issues on storage

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt [hunk 1]
```diff
@@ -275,8 +277,13 @@ class BookmarkFragment : LibraryPageFragment<BookmarkNode>(), BackHandler, Accou
     }
 
     private suspend fun deleteSelectedBookmarks(selected: Set<BookmarkNode>) {
-        selected.forEach {
-            context?.bookmarkStorage()?.deleteNode(it.guid)
+        CoroutineScope(IO).launch {
+            val tempStorage = context?.bookmarkStorage()
+            selected.forEach {
+                runBlocking {
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt]: It is necessary to delete each bookmark one at a time? Maybe you could use Deferred and [awaitAll](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/await-all.html).

Comment (sblatz) [reply app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt]: @mcarare any thoughts on trying this? :)

Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt]: You mean something like this: 

```
CoroutineScope(IO).launch {
            val tempStorage = context?.bookmarkStorage()
            val deferreds = selected.map {
                async {
                    tempStorage?.deleteNode(it.guid)
                }
            }
            deferreds.awaitAll()
        }
```

Comment (sblatz) [reply app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt]: Yeah that seems reasonable to me 😄

**Comment by mcarare** [comment]
Closing this, opening a new PR https://github.com/mozilla-mobile/fenix/pull/5559 with a clean branch to get rid of merging conflicts and test failures.

**Pull request closed without merging**